src/components/Home.js

```diff
@@ -20,29 +22,74 @@ const Home = () => {
     setExistingToken(event.target.value);
   };
 
-  const [shoppingList] = useCollectionData(
-    db.where('token', '==', existingToken),
-  );
-
   const submitToken = (e) => {
     e.preventDefault();
 
-    if (existingToken === '') {
-      alert('Please enter a token...');
+    if (existingToken.trim() === '') {
+      setShowModal(true);
+      setModalMessage('Please enter a token');
```

Comment:
There seems to be a difference in loading when I do not have a token and click join the list and inserting a wrong token. The first one it seems, there is no loading, and the modal just appears, while by the latter the whole page reloads.